homeassistant/components/plex/__init__.py

```diff
@@ -54,148 +55,94 @@
 
 CONFIG_SCHEMA = vol.Schema({PLEX_DOMAIN: SERVER_CONFIG_SCHEMA}, extra=vol.ALLOW_EXTRA)
 
-CONFIGURING = "configuring"
 _LOGGER = logging.getLogger(__package__)
 
 
 def setup(hass, config):
     """Set up the Plex component."""
 
-    def server_discovered(service, info):
-        """Pass back discovered Plex server details."""
-        if hass.data[PLEX_DOMAIN][SERVERS]:
-            _LOGGER.debug("Plex server already configured, ignoring discovery.")
-            return
-        _LOGGER.debug("Discovered Plex server: %s:%s", info["host"], info["port"])
-        setup_plex(discovery_info=info)
-
-    def setup_plex(config=None, discovery_info=None, configurator_info=None):
-        """Return assembled server_config dict."""
+    def setup_plex(config):
```

Comment:
We should be able to unnest this function now, if we pass in `hass` as first argument, besides `config`.